jgrapht-core/src/main/java/org/jgrapht/alg/partition/BipartitePartitionFinder.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
```

Comment:
Here, the main author's name as well and the date (see `git blame`).  When for example I refactor Barak's or John's code, unless I rewrite it completely, I never change the copyright.